pkg/ibmcsidriver/controller_test.go

```diff
@@ -446,15 +446,15 @@ func TestCreateVolumeArguments(t *testing.T) {
 				VolumeCapabilities: stdVolCap,
 				Parameters:         stdParams,
 			},
-			expErrCode:                codes.Internal,
+			expErrCode:                codes.InvalidArgument,
 			expVol:                    nil,
 			subnetID:                  "sub-1",
 			securityGroupID:           "kube-fake-cluster-id",
 			subnetError:               nil,
 			securityGroupError:        nil,
 			libVolumeResponse:         nil,
 			libVolumeAccessPointError: nil,
-			libVolumeError:            providerError.Message{Code: "FailedToPlaceOrder", Description: "Volume creation failed", Type: providerError.Unauthenticated},
+			libVolumeError:            providerError.Message{Code: "FailedToPlaceOrder", Description: "Volume creation failed", Type: providerError.Unauthenticated, RC: 400},
```

Comment:
Unauthenticated error looks to be not matching with error code codes.InvalidArgument

Reply:
https://github.ibm.com/alchemy-containers/armada-storage/issues/6171#issuecomment-106016766

Reply:
IF we are expecting this then we need to match exact error codes and right if else conditions or similar logic. As of now for this phase anything that is not internal is 4xx invalid argument. We can discuss if required.

Reply:
We are considering 2 generic categories, 5xx server side error and 4xx client is not passing right parameters

Reply:
still same comment code is not matching